sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -45,6 +45,7 @@
 from apache_beam.typehints import Union
 from apache_beam.typehints import with_input_types
 from apache_beam.typehints import with_output_types
+from apache_beam.typehints.decorators import _check_instance_type
```

Comment:
Functions whose names start with underscore are intended to be used only internally. So we shouldn't import _check_instance_type from another package.

Reply:
I tried using the decorator `with_input_types`. But, it was comparing the typehint with `WindowedValue`, whereas, the type of the element in the DoFn was a `tuple` which was actually the `WindowedValue` object's value attribute.

Reply:
`with_input_types` won't work here. Are there other public functions that does the check available? I still believe importing private functions is not a good approach.

Reply:
There are no public functions that do the same/something similar, except for `with_input_types` AFAIK. 

I think the problem here is that we want to check the type of the element which is different at pipeline construction-time and runtime.
While creating the PCollection, the type of the elements is `WindowedValue` and inside the DoFn, the type of the element is `Tuple`. 

If the input is an object of WindowedValue, any DoFn will have its element param as the value attribute of the WindowedValue.  I think the type evaluated at pipeline construction-time should also be the same. 
If that is the case, then `with_input_types` will work as intended here.